dbms/src/Flash/Mpp/MPPHandler.cpp

```diff
@@ -44,7 +46,9 @@ void MPPHandler::handleError(const MPPTaskPtr & task, String error)
 grpc::Status MPPHandler::execute(const ContextPtr & context, mpp::DispatchTaskResponse * response)
 {
     MPPTaskPtr task = nullptr;
-    current_memory_tracker = nullptr; /// to avoid reusing threads in gRPC
+    SCOPE_EXIT({
+        current_memory_tracker = nullptr; /// to avoid reusing threads in gRPC
```

Comment:
why a `SCOPE_EXIT` is added? since the meanings are quiet different

Reply:
Because after `MPPTask::prepare()`, `current_memory_tracker` will be set to the memory tracker of the `MPPTask`, and if we don't set it back to nullptr, then this thread may have illegal memory tracker.